cmd/kubeadm/app/cmd/phases/bootstraptoken.go

```diff
@@ -17,317 +17,112 @@ limitations under the License.
-		},
+func runBoostrapToken(c workflow.RunData) error {
+	data, ok := c.(bootstrapTokenData)
+	if !ok {
+		return errors.New("bootstrap-token phase invoked with an invalid data struct")
 	}
-
-	// Adds flags to the command
-	addGenericFlags(cmd.Flags(), &cfgPath, &skipTokenPrint)
-	bto.AddTokenFlag(cmd.Flags())
-	bto.AddTTLFlag(cmd.Flags())
-	bto.AddUsagesFlag(cmd.Flags())
-	bto.AddGroupsFlag(cmd.Flags())
-	bto.AddDescriptionFlag(cmd.Flags())
-
-	return cmd
-}
-
-// NewSubCmdClusterInfo returns the Cobra command for running the cluster-info sub-phase
-func NewSubCmdClusterInfo(kubeConfigFile *string) *cobra.Command {
-	cmd := &cobra.Command{
-		Use:     "cluster-info",
-		Short:   "Uploads the cluster-info ConfigMap from the given kubeconfig file",
-		Long:    clusterInfoLongDesc,
-		Aliases: []string{"clusterinfo"},
-		Run: func(cmd *cobra.Command, args []string) {
-			client, err := kubeconfigutil.ClientSetFromFile(*kubeConfigFile)
-			kubeadmutil.CheckErr(err)
-
-			// Create the cluster-info ConfigMap or update if it already exists
-			err = clusterinfo.CreateBootstrapConfigMapIfNotExists(client, *kubeConfigFile)
-			kubeadmutil.CheckErr(err)
-
-			// Create the RBAC rules that expose the cluster-info ConfigMap properly
-			err = clusterinfo.CreateClusterInfoRBACRules(client)
-			kubeadmutil.CheckErr(err)
-		},
+	cfg := data.Cfg()
+	skipTokenPrint := data.SkipTokenPrint()
+	kubeconfigPath := data.KubeConfigPath()
+	client, err := data.Client()
+	if err != nil {
+		return err
 	}
-	return cmd
-}
 
-// NewSubCmdNodeBootstrapToken returns the Cobra command for running the node sub-phase
-func NewSubCmdNodeBootstrapToken(kubeConfigFile *string) *cobra.Command {
-	cmd := &cobra.Command{
-		Use:     "node",
-		Short:   "Configures the node bootstrap process",
-		Aliases: []string{"clusterinfo"},
-		Long:    cmdutil.MacroCommandLongDescription,
+	tokens := []string{}
+	for _, bt := range cfg.BootstrapTokens {
+		tokens = append(tokens, bt.Token.String())
 	}
-
-	cmd.AddCommand(NewSubCmdNodeBootstrapTokenPostCSRs(kubeConfigFile))
-	cmd.AddCommand(NewSubCmdNodeBootstrapTokenAutoApprove(kubeConfigFile))
-
-	return cmd
-}
-
-// NewSubCmdNodeBootstrapTokenPostCSRs returns the Cobra command for running the allow-post-csrs sub-phase
-func NewSubCmdNodeBootstrapTokenPostCSRs(kubeConfigFile *string) *cobra.Command {
-	cmd := &cobra.Command{
-		Use:   "allow-post-csrs",
-		Short: "Configures RBAC to allow node bootstrap tokens to post CSRs in order for nodes to get long term certificate credentials",
-		Long:  nodePostCSRsLongDesc,
-		Run: func(cmd *cobra.Command, args []string) {
-			client, err := kubeconfigutil.ClientSetFromFile(*kubeConfigFile)
-			kubeadmutil.CheckErr(err)
-
-			// Create RBAC rules that makes the bootstrap tokens able to post CSRs
-			err = node.AllowBootstrapTokensToPostCSRs(client)
-			kubeadmutil.CheckErr(err)
-		},
+	if !skipTokenPrint {
+		if len(tokens) == 1 {
+			glog.V(1).Infof("[bootstraptoken] using token: %s\n", tokens[0])
```

Comment:
please use printf
[bootstraptoken] --> [bootstrap-token] (text in square breakets should match the phase name)
creating --> Creating (start all message with capital letter)

same for all the printf messages

Reply:
@yagonobre we are using fmt.printf for things we want to notify to the user always (the default kubeadm output) and glog for info/debug message that can be triggered with the -v flag